### PR TITLE
FIX-#2239: Compute row index start using pandas

### LINKED ISSUE #2239: Row index does not match pandas on read_csv with skiprows and names on certain files
pandas has a very strange edge case here. The following conditions must be met:

* `skiprows` must be > 1
* `names` must not be `None`
* `index_col` must be `None`
* File must have different data types in each line above `skiprows` number

Since this is data dependent, the best course of action is to hand the decision of what number to start to pandas.

### PR DESCRIPTION
**Pull request by devin-petersohn**
Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2239 
- [x] tests added and passing

### CODE REVIEW

File: modin/engines/base/io/text/csv_reader.py [hunk 1]
```diff
@@ -171,10 +171,17 @@ def _read(cls, filepath_or_buffer, **kwargs):
...
+            # If skiprows and names are specified, pandas assigns a row number based
+            # on the number of dtypes that match above.
+            # This number is not easy for us to compute and ensure matching behavior
+            # with pandas, so we will just read 1 line with pandas and grab the start
+            # value from that.
+            if skiprows > 1 and kwargs.get("names", None) is not None:
+                start = pandas.read_csv(
+                    filepath_or_buffer, skiprows=skiprows, nrows=1, names=names
+                ).index[0]
+                new_index = pandas.RangeIndex(start, start + new_index.stop)
```
Comment (dchigarev) [inline modin/engines/base/io/text/csv_reader.py:184]: if I understood it correctly, when we hit our described corner case, then start would be `np.nan`, so `start + new_index.stop` will also be NaN.

I'm wondering how tests passing...

Comment (devin-petersohn) [reply modin/engines/base/io/text/csv_reader.py:184]: It should never be `np.nan` did you encounter this in testing?

Comment (dchigarev) [reply modin/engines/base/io/text/csv_reader.py:184]: oh, sorry, I thought it was fixing #1930))
but what cases then it fixing? your new added tests pass even on the current master, without your changes in this PR

Comment (devin-petersohn) [reply modin/engines/base/io/text/csv_reader.py]: The fix is simpler, I struggled to recreate the issue because our assumptions in the code are no longer true. The latest changes add a test that fails on master and is fixed by removing the code that produces the incorrect index.

**Review by dchigarev — APPROVED** [review]
LGTM

File: modin/engines/base/io/text/csv_reader.py (hunk 1 above)
Comment (dchigarev) [reply modin/engines/base/io/text/csv_reader.py]: Okay, then it seems that it's ready to merge
